src/lib.rs

```diff
@@ -315,24 +422,35 @@ macro_rules! impl_trait(
                 }
             }
         }
+
+        #[cfg(feature = "alloc")]
+        unsafe impl FromByteVec for $to {
+            fn from_byte_vec(mut vec: Vec<u8>) -> Result<Vec<$to>, Error> {
+                let (len, capacity) = check_vec_constraints::<$to>(&vec)?;
+                let ptr = vec.as_mut_ptr();
+                mem::forget(vec);
+                unsafe {
+                    Ok(Vec::from_raw_parts(ptr as *mut $to, len, capacity))
+                }
+            }
+        }
+
+        #[cfg(feature = "alloc")]
+        impl IntoByteVec for Vec<$to> {
+            fn into_byte_vec(mut self) -> Vec<u8> {
+                let size = mem::size_of::<$to>();
+                let len = self.len() * size;
+                let capacity = self.capacity() * size;
+                let ptr = self.as_mut_ptr();
+                mem::forget(self);
+                unsafe {
+                    Vec::from_raw_parts(ptr as *mut u8, len, capacity)
```

Comment:
Also here special casing for empty vecs

Reply:
Got it :)

Reply:
I made the change, we are returning empty vec early just to avoid additional casting right? or would it fail for a particular edge case?